packages/core-api/src/extensions/ErrorBoundary.tsx

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+
+export type ErrorBoundaryFallbackProps = {
+  error: Error;
+  resetErrorBoundary: () => void;
+};
+
+type FallbackRender = React.FunctionComponent<ErrorBoundaryFallbackProps>;
+
+type Props = {
+  fallbackRender: FallbackRender;
```

Comment:
And `FallbackComponent` here. Components can be inlined as functions now anyway, so render functions are a bit redundant

Reply:
WDYM here?

If the type of this becomes a component, then the component class type fails with "has no call signatures".

Reply:
I'm thinking you'd mirror how it's set up here: https://github.com/carloscuesta/react-native-error-boundary/blob/master/src/index.d.ts